### PR TITLE
add null checks to make strictNullChecks happy

### PR DESCRIPTION
**Pull request by ruemic**
Was getting yelled at my compiler
screen shot 2016-09-11 at 10 58 18 pm

### CODE REVIEW

File: src/components/creeps/creepAction.ts [hunk 1]
```diff
@@ -27,7 +27,9 @@ export default class CreepAction implements ICreepAction {
 
   public setCreep(creep: Creep) {
     this.creep = creep;
-    this.renewStation = Game.getObjectById<Spawn>(this.creep.memory.renew_station_id);
+    let spawn = Game.getObjectById<Spawn>(this.creep.memory.renew_station_id);
+    if (spawn == null) { return false; };
```
Comment (resir014) [inline src/components/creeps/creepAction.ts]: - `return false;` should be on its own line, and remove the trailing semicolon outside of the curly brace, e.g.

``` js
if (spawn == null) {
  return false;
}
```
- The following null check has double equals while the others have triple equals. Would be nice if we could use double/triple equals consistently.

File: src/components/creeps/harvester.ts [hunk 2]
```diff
@@ -21,9 +21,11 @@ export default class Harvester extends CreepAction implements IHarvester, ICreep
 
   public setCreep(creep: Creep) {
     super.setCreep(creep);
-
-    this.targetSource = Game.getObjectById<Source>(this.creep.memory.target_source_id);
-    this.targetEnergyDropOff = Game.getObjectById<Spawn | Structure>(this.creep.memory.target_energy_dropoff_id);
+    let source = Game.getObjectById<Source>(this.creep.memory.target_source_id);
+    let dropOff = Game.getObjectById<Spawn | Structure>(this.creep.memory.target_energy_dropoff_id);
+    if (source === null || dropOff === null) { return false; };
```
Comment (resir014) [inline src/components/creeps/harvester.ts]: - `return false;` should be on its own line, and remove the trailing semicolon outside of the curly brace, e.g.

``` js
if (source === null || dropOff === null) {
  return false;
}
```

**Comment by resir014** [comment]
Alright, I've added a few line comments to make sure the coding styles are consistent with the rest. Will merge when they're fixed.

**Comment by ruemic** [comment]
@resir014 cleaned up & squashed

**Comment by resir014** [comment]
Thanks! Merged.

**Pull request merged**